### PR TITLE
Raise specification after section

### LINKED ISSUE #118: Ordering issue in VRPLIB files
This:
```
EDGE_WEIGHT_SECTION
0     1544  1944  1931  1476
1726     0  1992  1427  1593
1965  1975     0  621   1090
2063  1433   647  0     818
1475  1594  1090  828   0
EDGE_WEIGHT_TYPE : EXPLICIT
EDGE_WEIGHT_FORMAT : FULL_MATRIX
```
has some of the specifications after the edge weight data. This does not currently work, because VRPLIB parses the `EDGE_WEIGHT_SECTION` the moment it encounters it and assumes all relevant specifications are known by that time. Clearly, here, they are not, and that causes an error. I don't think it should, but if it should, we should definitely clarify that we explicitly assume the specifications come first in the data file.

I ran into this when I was rewriting the GVRP instances yesterday.

### PR DESCRIPTION
**Pull request by leonlan**
This PR raises a ValueError when a specification is introduced after a section. Closes #118.

### CODE REVIEW

**Comment by leonlan** [comment]
This is probably the last thing I want to include with the current parser because it's getting too messy 😅. I'll have a go at a VRPLIB grammar soon.

**Comment by N-Wouda** [comment]
This also closes #118, right?

File: vrplib/parse/parse_vrplib.py [hunk 1]
```diff
@@ -79,6 +77,9 @@ def group_specifications_and_sections(lines: list[str]):
             end_section = start + 1
 
             for next_line in lines[start + 1 :]:
+                if ":" in next_line:
```
Comment (N-Wouda) [inline vrplib/parse/parse_vrplib.py:80]: Is this a sufficient test for a specification? (probably yes but can there ever be data in a section with a colon in it?)

**Review by N-Wouda — APPROVED** [review]

File: vrplib/parse/parse_vrplib.py (hunk 1 above)
Comment (leonlan) [reply vrplib/parse/parse_vrplib.py:80]: I cannot think of any data section that must include `:`'s. Only thing I can think of are strings but we should probably avoid strings in data sections as they can often be mapped to numerical values. I think we're OK if we define the VRPLIB standard to reserve `:` as a specifications identifier.

**Comment by leonlan** [comment]
> This also closes #118, right?

Yea, linked it to the PR!

File: vrplib/parse/parse_vrplib.py (hunk 1 above)
Comment (N-Wouda) [reply vrplib/parse/parse_vrplib.py:80]: I am excited about having a grammar for this so we can definitively say what's OK and what's not 😆.

**Pull request merged**